apache/zeppelin
# [ZEPPELIN-1372]Automatically Detect the data type in table and sort the numbers correctly

### What is this PR for?
Automatically detect the data type for the contents in the table.
This enables the sorting on numerical/date columns.
Please see https://issues.apache.org/jira/browse/ZEPPELIN-1372 for details.


### What type of PR is it?
[Bug Fix | Improvement]

### Todos

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-1372

### How should this be tested?
Click on the title of columns and see whether the numeric contents are correctly sorted.

### Screenshots (if appropriate)
#### For numbers
Before
![screen shot 2016-08-26 at 1 35 02 pm](https://cloud.githubusercontent.com/assets/3334391/18019657/f6669852-6b91-11e6-9ff6-6bbde68bce20.png)
After
![screen shot 2016-08-26 at 1 34 00 pm](https://cloud.githubusercontent.com/assets/3334391/18019661/fa6dcd8a-6b91-11e6-95ce-339218b53e5a.png)

#### For dates
Before
![screen shot 2016-08-30 at 3 24 02 pm](https://cloud.githubusercontent.com/assets/3334391/18109384/df615a7a-6ec5-11e6-9c8f-6f07f5f56bd4.png)
After
![screen shot 2016-08-30 at 3 11 14 pm](https://cloud.githubusercontent.com/assets/3334391/18109351/9f28c48e-6ec5-11e6-9184-87093fcac5fe.png)

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No
